### PR TITLE
 GPUGridLayer PART-2: Add support for multiple weights

### PR DESCRIPTION
**Pull request by 1chandu**
For #2084 

#### Background
 - Update gpu aggregation utility methods  to support multiple weights.
 - Update Contour and GPUGridLayers to use these updated methods.
 - Update `GPUGridCellLayer` (private to GPUGridLayer) JS and Shaders to support color and elevation, based on aggregation results.
 - Add unit test for GPUGridCell layer's vertex shader.

**NEXT**: RFC and PR to integrate `GPUGridLayer` with `GridLayer`.

#### Change List
- GPU Grid Aggregation PART-2: Add support for multiple weights

### CODE REVIEW

File: modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js [hunk 1]
```diff
@@ -120,26 +154,35 @@ export default class GPUGridLayer extends CompositeLayer {
...
     } = this.props;
 
-    const {countsBuffer, maxCountBuffer, gridSize, gridOrigin, cellSize} = this.state;
+    const {weights, gridSize, gridOrigin, cellSize} = this.state;
     const minColor = MINCOLOR;
     const maxColor = MAXCOLOR;
 
+    const colorRange = [];
+    this.props.colorRange.forEach(color => {
+      colorRange.push(color[0], color[1], color[2], color[3] || 255);
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js]: Always assign into a pre-allocated typed array. Push has perf issues.

File: modules/aggregation-layers/src/utils/gpu-grid-aggregation/grid-aggregation-utils.js [hunk 2]
```diff
@@ -108,15 +108,18 @@ function parseGridData(data, getPosition, getWeight = null) {
...
-    // Aggregator expects each weight is an array of size 3
-    if (Array.isArray(weight)) {
-      weightValues[index * 3] = weight[0];
-      weightValues[index * 3 + 1] = weight[1];
-      weightValues[index * 3 + 2] = weight[2];
-    } else {
-      // backward compitability
-      weightValues[index * 3] = weight;
+    for (const name in weightParams) {
+      const weight = weightParams[name].getWeight(object);
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/utils/gpu-grid-aggregation/grid-aggregation-utils.js]: Is this a breaking change?

Comment (1chandu) [reply modules/aggregation-layers/src/utils/gpu-grid-aggregation/grid-aggregation-utils.js]: this is a private API used internally by layers within `aggregation-layers` module, I just double checked and removed an export (was an experimental) from `aggregation-layers`.

File: modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js (hunk 1 above)
Comment (1chandu) [reply modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js]: Fixed.

File: modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js [hunk 3]
```diff
@@ -120,26 +154,39 @@ export default class GPUGridLayer extends CompositeLayer {
...
     const minColor = MINCOLOR;
     const maxColor = MAXCOLOR;
 
+    const colorRange = new Float32Array(this.props.colorRange.length * 4);
+    this.props.colorRange.forEach((color, index) => {
+      const colorRangeIdnex = index * 4;
+      colorRange[colorRangeIdnex] = color[0];
+      colorRange[colorRangeIdnex + 1] = color[1];
+      colorRange[colorRangeIdnex + 2] = color[2];
+      colorRange[colorRangeIdnex + 3] = color[3] || 255;
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js]: `color[3]` can be `0`

File: modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-cell-layer.js [hunk 4]
```diff
@@ -53,9 +54,14 @@ export default class GPUGridCellLayer extends Layer {
   initializeState() {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-cell-layer.js]: Should this layer be extending the base `GridCellLayer`?

Comment (1chandu) [reply modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-cell-layer.js]: Attribute setup, state change detection, uniforms and shaders are completely different for both `GridCellLayer` and `GPUGridCellLayer` hence not extending from `GridCellLayer`.

File: modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js (hunk 3 above)
Comment (1chandu) [reply modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-layer.js]: Thanks ! , fixed in couple of places.

File: modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-cell-layer.js (hunk 4 above)
Comment (1chandu) [reply modules/aggregation-layers/src/gpu-grid-layer/gpu-grid-cell-layer.js]: I will port recent elevation calculation changes from Column layer to GPUGridCellLayer in a separate PR.

**Pull request merged**